### PR TITLE
An/3286

### PR DESCRIPTION
**Pull request by angwangri**
### Description:

- On the accounts screen, I added a condition. 
If there is no nickname, it shows the truncated address.
If there is a nickname, it shows nickname without an address.
- On the account detail page, it shows the full address. 
- Tapping on the address in the account detail screen will show a popup that allows copy.

### Checklist:

- [ ] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

**Comment by franckc** [comment]
@angwangri Thanks for submitting this PR ! :) When it's ready for review, please add a short description in the section above, then ping me and I'll find reviewers for it.

**Comment by franckc** [comment]
@tomlinton @sparrowDom Do you mind reviewing this PR since it deals with mobile code and you are experts ? Thanks ! :)

File: mobile/src/utils/index.js [hunk 1]
```diff
@@ -36,13 +36,19 @@ export function evenlySplitAddress(address = '') {
...
+    return address
 
   const separator = '...'
+  const firstPortionLength = Math.floor(chars / 2)
   return (
-    address.substr(0, chars) +
+    address.substr(0, firstPortionLength) +
     separator +
-    address.substr(address.length - chars)
+    address.substr(address.length - Math.max(1, chars - 3 - firstPortionLength))
```
Comment (sparrowDom) [inline mobile/src/utils/index.js]: nitpick: I know I suggested this part, taking a second look perhaps `separator.length` is more explanatory than `3`

**Comment by sparrowDom** [comment]
@angwangri sorry for not being clear enough with my comments on the previous PR. I think the changes to `settings.js` and `changePin.js` are not part of this pull request (rather the should only be in this one: https://github.com/OriginProtocol/origin/pull/3688 )?

File: mobile/src/utils/index.js (hunk 1 above)
Comment (angwangri) [reply mobile/src/utils/index.js]: i think that it should be :
address.substr(address.length - Math.max(3, chars - 3 - firstPortionLength))

is it correct ?

**Comment by angwangri** [comment]
Okay, I revert changePin and setting files.

File: mobile/src/utils/index.js (hunk 1 above)
Comment (sparrowDom) [reply mobile/src/utils/index.js]: Yeah I think you are right. Also for clarity I would upgrade it to
```
address.substr(address.length - Math.max(3, chars - separator.length - firstPortionLength))
```
to make a better sense of where the "magic 3" is coming from :)

**Comment by angwangri** [comment]
@sparrowDom I fixed all issues. Can you check again ?

**Comment by sparrowDom** [comment]
nice! lgtm

**Pull request merged**